tests/base.py

```diff
@@ -210,6 +210,18 @@ def select_all_streams_and_fields(conn_id, catalogs, select_all_fields: bool = T
             connections.select_catalog_and_fields_via_metadata(
                 conn_id, catalog, schema, additional_md, non_selected_properties)
 
+    def unique_pk_count_by_stream(self, recs_by_stream):
+        """
+        Switch from upsert record count verification to unique pk count verification due to
+        tap-mssql inconsistency with log based inclusivity TDL-24162 (that will not be fixed)
+        """
+        pk_count_by_stream = {}
+        for strm in recs_by_stream:
+            stream_pks = [m.get('data').get('pk') for m in recs_by_stream[strm]['messages']
```

Comment:
```suggestion
            stream_pks = [m.get('data').get('pk') for m in strm['messages']
```
You're already at the stream level, no need to go up to the dictionary-of-all-streams level just to come back down

Reply:
I thought I tried this first and it was just the stream instead of the stream and everything under it.  I'll have another look.

Reply:
After a second pass strm is indeed just a string with the top level key of the recs_by_stream dict.  It is not unpacking the underlying data.  So I can't then use strm['messages'].  Let me know if there's a way around this:
```
> /opt/code/tap-mssql/tests/base.py(222)unique_pk_count_by_stream()
    221             1+1
--> 222             stream_pks = {m.get('data', {}).get('pk') for m in strm['messages']
    223                           if m['action'] == 'upsert'}

ipdb> type(strm)
<class 'str'>
ipdb> type(recs_by_stream)
<class 'dict'>
ipdb> recs_by_stream.keys()
dict_keys(['data_types_database_dbo_1834871389834_start_with_numbers', 'data_types_database_dbo_hebrew_ישראל', "data_types_database_dbo_invalid_characters_ !#$%&'()*+,-./:;<=>?@[\\]^_`{|}~", 'data_types_database_dbo_SELECT'])
ipdb> strm
'data_types_database_dbo_1834871389834_start_with_numbers'
ipdb>
```

Reply:
HA! I forgot how write python I guess.

```python
recs_by_stream = {
    "stream1": {"messages": [1, 2, 3]},
    "stream2": {"messages": [4, 5, 6]},
}

for stream, recs in recs_by_stream.items():
    print(f"{stream} {recs}")
```

Results:
```
stream1 {'messages': [1, 2, 3]}
stream2 {'messages': [4, 5, 6]}
```